### ISSUE TITLE
Developer: app-to-app put request has no callback

### ISSUE THREAD

**Issue by irritanterik** [athombv/homey#714]
**What is your Homey version (Settings → System)?**
0.8.39

**What did you try to achieve?**
app-to-app put request from Http app to Better Logic app
while get requests are handled correctly, a callback is missing on put requests.
Without a callback on put requests error handling cannot be done optimally.

_Please include detailed steps how we can reproduce the issue_
Create a string variable in BetterLogic named 'puttest'.
Send request from App like:

```
var api = Homey.manager('api')
var betterLogic = new api.App('net.i-dev.betterlogic')
betterLogic.put('/puttest/newValue', function(err) { console.log(err)})
```

**What was the result?**
Callback function never called.
...

**What did you expect as result?**
A callback with success/failure indication
...

**Comment by WeeJeWel** [comment]
You're missing the 2nd parameter, your PUT body, e.g. `{}` or `null`.

**Comment by irritanterik** [comment]
Thanks

**Comment by irritanterik** [comment]
@jeroenvollenbrock Please reopen this issue. Adding a body (2nd parameter) has no result. The third parameter (callback function) is never called on app-to-app PUT requests.

**Comment by WeeJeWel** [comment]
Definitely fixed in 0.10, probably before.

**Comment by irritanterik** [comment]
@WeeJeWel Please reopen this issue.
With this snippet (http app) callback is never called:
```
    var api = Homey.manager('api')
    var betterLogic = new api.App('net.i-dev.betterlogic')
    
    betterLogic.put('/' + variableName + '/' + someValue, null, function (err, result) {
      if (err) return Homey.log('error', err)
      Homey.log('ok', result)
    })

```

**Comment by irritanterik** [comment]
Any update?

**Comment by irritanterik** [comment]
Any update?

**Comment by WeeJeWel** [comment]
Have you tried recently? This should work :)

**Comment by irritanterik** [comment]
Just now on 1.0.5
```
    betterLogic.put('/' + args.betterVariable.name + '/' + variable, null, function (err, result) {
      // https://github.com/athombv/homey/issues/714
      if (err) return callback(err)
      callback(null, true)
    })
```

**Comment by WeeJeWel** [comment]
I'll look into it tomorrow!

**Comment by WeeJeWel** [comment]
Whoops, you were right! Fixed in v1.1.0

**Comment by irritanterik** [comment]
Confirmed!
